x/shield/keeper/pool.go

```diff
@@ -32,16 +32,20 @@ func (k Keeper) CreatePool(
 		return types.Pool{}, types.ErrNotShieldAdmin
 	}
 
+	if _, err := k.GetPoolBySponsor(ctx, sponsor); err == nil {
```

Comment:
Not a real problem, but maybe it would be better if we can return `found` instead of `error`.